go/WorkQueue.go

```diff
@@ -120,6 +123,85 @@ func (workQueue *WorkQueue) AddItem(ctx context.Context, db *redis.Client, item
 	return err
 }
 
+// **************-This method can be used only when database is locked-**************
+func (workQueue *WorkQueue) AddAtomicItem(ctx context.Context, db *redis.Client, item Item) error {
+	txf := func(tx *redis.Tx) error {
+		pipe := db.Pipeline()
+
+		mainQueueLenCmd := pipe.LLen(ctx, workQueue.mainQueueKey)
+		processingQueueLenCmd := pipe.LLen(ctx, workQueue.processingKey)
```

Comment:
How come we need to get the lengths?

Reply:
[in ](https://pkg.go.dev/github.com/go-redis/redis/v8#Client.LPos)
https://pkg.go.dev/github.com/go-redis/redis/v8#LPosArgs

Reply:
It only uses the max length if it isn't 0: https://github.com/redis/go-redis/blob/v8.11.5/commands.go#L1422 (the library has poor documentation).

Therefore setting it to 0 won't pass the argument, and therefore the whole list will be checked (https://redis.io/commands/lpos/).

In Go, we can just not pass the parameter when creating the struct and it'll default to 0. Similarly, you don't need to specify rank, either.